ui/app/serializers/connector.js

```diff
@@ -27,8 +27,86 @@ export default class ConnectorSerializer extends ApplicationSerializer {
 
   extractRelationship(modelName, value) {
     if (modelName === 'plugin' && value) {
+      const pluginNames = this.store.peekAll('plugin').mapBy('name');
```

Comment:
Would it make sense to add a unit test for either the `extractRelationship` method of the serializer or an acceptance test asserting the correct display of connector labels in the UI?

This way, we could not only guard check against regressions, but also make it easier to add new rules for the labelling in the future